src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -441,7 +442,7 @@ private static ModuleMetadata ReadModuleMetadataFrom(ObjectReader reader, Serial
         {
             Contract.ThrowIfFalse(SerializationKinds.Bits == kind);
 
-            var array = reader.ReadArray<byte>();
+            var array = reader.ReadByteArray();
```

Comment:
specialzied entrypoints for char[] and byte[] the only two actual array types we ever need to write out.